### ISSUE TITLE
aws_s3 sink: broken in version 0.32.1-*

### ISSUE THREAD

**Issue by cdancy** [vectordotdev/vector#18548]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Problem

Version `0.31.0-*` works as expected. When we upgrade to version `0.32.1-*` we see the attached pics (effectively a panic is triggered) . Our config for the sink looks as follows:
```
  http_server_logs_to_s3:
    bucket: my-hello-world-bucket
    type: aws_s3
    key_prefix: "hello-world/"
    encoding:
      codec: json
    inputs:
    - http_server_logs
    auth:
      assume_role: "arn:aws:iam::123456789:role/hello-world-cluster-EKSLoggingIRSARole"
    batch:
      timeout_secs: 60
```
<img width="1337" alt="Screenshot 2023-09-12 at 1 06 50 PM" src="https://github.com/vectordotdev/vector/assets/7871459/54491d48-5575-4f6c-9d0f-78b83ec268b9">

<img width="1326" alt="Screenshot 2023-09-12 at 2 49 11 PM" src="https://github.com/vectordotdev/vector/assets/7871459/5abe71f2-42c1-4580-b1c3-16b341edfdc7">

### Version

0.32.1

**Comment by jszwedko** [comment]
Hi @cdancy !

Sorry you ran into this. It was actually reported before in https://github.com/vectordotdev/vector/issues/18340 which we did fix in `master` and plan to release in `0.32.2` soon.

**Comment by cdancy** [comment]
@jszwedko aahhh ... my "search kung-fu" missed that :) Thanks for the update!
